lib/matplotlib/backend_bases.py

```diff
@@ -3428,7 +3422,13 @@ def add_tool(self, name, tool, group=None, position=None):
                           'not added')
             return
 
-        self._tools[name] = tool_cls(self, name)
```

Comment:
Sure, but what do you think of the main-use case, someone calling:
```python
my_tool = MyTool(None, 'foo')
navigation.add_tool('foo', my_tool)
```

My thoughts went along the lines of we allow this, then we want to make sure that the navigation gets set to this one.

If you think the user should do this, then could make make navigation optional until we make it so in ``navigation.add_tool``.  I don't mind either way about whether we allow a tool object to get passed in.